builders/build/substrate-api/polkadot-js-api.md

```diff
@@ -0,0 +1,301 @@
+
+### Checking Prerequisites {: #checking-prerequisites }
+
+Installing and using polkadot.js API library requires node.js to be installed. 
+
+--8<-- 'text/common/install-nodejs.md'
+
+#### Installing Polkadot.js API library {: #installing-polkadot.js-api-library } 
+
+To get started with the polkadot.js library, install it in your project directory using the following command:
+
+```
+yarn add @polkadot/api
+```
+
+#### Installing Moonbeam Types Bundle {: #moonbeam-types-bundle }
+
+For decoding Moonbeam custom events and types, you will need to include the [Moonbeam Types Bundle](https://www.npmjs.com/package/moonbeam-types-bundle){target=blank} into your project by adding the following package information to your `package.json`:
+
+```json
+"@polkadot/api": "^6.9.1",
+"moonbeam-types-bundle": "^2.0.1",
+"typescript": "4.3.2"
+```
+
+And add this import statement to the start of your project file:
+
+```javascript
+import { typesBundlePre900 } from "moonbeam-types-bundle"
+```
+
+## Creating an API Provider Instance {: #creating-an-API-provider-instance }
+
+Similar to ETH API libraries, you must first instantiate an API instance of polkadot.js API. Create the WsProvider using the websocket endpoint of the Moonbeam network you wish to interact with. 
+
+=== "Moonbeam"
+    ```javascript
+    // Import
+    import { ApiPromise, WsProvider } from '@polkadot/api';
+
+    // Construct
+    const wsProvider = new WsProvider('{{ networks.moonbeam.wss_url }}');
+    const api = await ApiPromise.create({ provider: wsProvider });
+
+    // Do something
+    console.log(api.genesisHash.toHex());
+    ```
+
+=== "Moonriver"
+    ```javascript
+    // Import
+    import { ApiPromise, WsProvider } from '@polkadot/api';
+
+    // Construct
+    const wsProvider = new WsProvider('{{ networks.moonriver.wss_url }}');
+    const api = await ApiPromise.create({ provider: wsProvider });
+
+    // Do something
+    console.log(api.genesisHash.toHex());
+    ```
+
+=== "Moonbase Alpha"
+    ```javascript
+    // Import
+    import { ApiPromise, WsProvider } from '@polkadot/api';
+
+    // Construct
+    const wsProvider = new WsProvider('{{ networks.moonbase.wss_url }}');
+    const api = await ApiPromise.create({ provider: wsProvider });
+
+    // Do something
+    console.log(api.genesisHash.toHex());
+    ```
+
+=== "Moonbeam Dev Node"
+    ```javascript
+    // Import
+    import { ApiPromise, WsProvider } from '@polkadot/api';
+
+    // Construct
+    const wsProvider = new WsProvider('{{ networks.development.wss_url }}');
+    const api = await ApiPromise.create({ provider: wsProvider });
+
+    // Do something
+    console.log(api.genesisHash.toHex());
+    ```
+
+
+## Querying for Information {: #querying-for-information }
+
+In this section, you will learn how to query for on-chain information using polkadot.js API library. 
+
+### Metadata and Dynamic API Decoration {: #metadata-and-dynamic-api-decoration }
```

Comment:
I think this section should be in the provider part. It is generic and not really applicable only to the queering part